programs/marinade-finance/src/instructions/crank/merge_stakes.rs

```diff
@@ -176,6 +177,9 @@ impl<'info> MergeStakes<'info> {
             .ok_or(MarinadeError::CalculationFailure)?
             .checked_sub(source_stake_info.last_update_delegated_lamports)
             .ok_or(MarinadeError::CalculationFailure)?;
+        // Note: since we're requiring source.delegation.stake == source.last_update_delegated_lamports
+        // and dest.delegation.stake == dest.last_update_delegated_lamports
+        // extra_delegated should be always ZERO
```

Comment:
NO. I saw some strange behavior when merging not activated stakes when dest.delegation.stake after merge = dest.last_update_delegated_lamports + source.last_update_delegated_lamports + rent_exempt. But the difference is < 0.01SOL so we can just increase mSOL price here